src/main/java/org/broadinstitute/hellbender/utils/runtime/ProcessControllerBase.java

```diff
@@ -0,0 +1,157 @@
+package org.broadinstitute.hellbender.utils.runtime;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public abstract class ProcessControllerBase<CAPTURE_POLICY extends CapturedStreamOutput>
+{
+    private static final Logger logger = LogManager.getLogger(ProcessControllerBase.class);
+
+    protected enum ProcessStream {Stdout, Stderr}
+
+    // Tracks running controllers.
+    protected static final Set<ProcessControllerBase<?>> running = Collections.synchronizedSet(new LinkedHashSet<>());
+
+    // We can't use a fixed thread pool size, since we could run out if multiple controllers were to make blocking calls
+    final static protected ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
we should probably provide a thread factory here, that A: sets the threads as daemon threads, B) gives them some sane name,

Reply:
Good idea - done.